### PR TITLE
`envoy.code.check`: Add upstream settings for extension metadata

### PR DESCRIPTION
**Pull request by phlax**
https://github.com/envoyproxy/envoy/issues/29305

### CODE REVIEW

**Comment by phlax** [comment]
cc @alyssawilk

File: envoy.code.check/envoy/code/check/typing.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class BaseExtensionMetadataDict(TypedDict):
     categories: List[str]
     security_posture: str
     status: str
+    status_upstream: str
```
Comment (alyssawilk) [inline envoy.code.check/envoy/code/check/typing.py:22]: I think we only want to require status_upstream if one of the categories has the string literal "http.upstream"

Comment (phlax) [reply envoy.code.check/envoy/code/check/typing.py:22]: cool ill check the check logic to see if we can accomodate - and add some commenting about per-cat status

Comment (phlax) [reply envoy.code.check/envoy/code/check/typing.py:22]: > I think we only want to require

i guess that goes for allow also

Comment (alyssawilk) [reply envoy.code.check/envoy/code/check/typing.py:22]: indeed!

**Pull request merged**